### DISCUSSION TITLE
Any thoughts on XetHub?

### DISCUSSION THREAD

**Discussion by julien-c** [git-lfs/git-lfs#5233] [Ideas] [answered]
Should lfs evolve to become more like this?

https://www.reddit.com/r/MachineLearning/comments/znfgap/p_xethub_we_scaled_git_to_support_1_tb_repos/

> XetHub: We scaled Git to support 1 TB repos

> Unlike Git LFS, XetHub doesn’t just store the files. It uses content-defined chunking and Merkle Trees to dedupe against everything in history, allowing small changes in large files to be stored compactly. Here’s how it works: https://xethub.com/assets/docs/how-xet-deduplication-works

**Comment by chrisd8088** [comment, marked as answer]
It's certainly an interesting approach, and the use of Git notes to store Merkle trees seems novel, to me at least.  I hope they choose to make their client code and APIs open and publicly available!
